### PR TITLE
Add new filters for other collections

### PR DESCRIPTION
**Pull request by brauliorivas**
BEGINRELEASENOTES
- Following GSoC tasks, we need to add more filters for other types of collections. That's why, depending on the view, filters will be added according to the collections in that view. 
- However, when first developing filters, I didn't take that well into account to allow for more filters. I think that the current code for filters is tangled and hard to follow. So I'm making it easier to understand and rewrite that part.
- The goal is to add filters for all the collections that eede currently supports. 

ENDRELEASENOTES
Currently, there are a few bugs when filtering objects in https://key4hep.github.io/eede/main/index.html because of #65. For example, when applying filters, the screen doesn't scroll to the new filtered objects. I'll also fix those bugs, and solve #53 because now is the moment.

### CODE REVIEW

**Comment by tmadlener** [comment]
I suppose since this is still WIP not everything is supposed to be working fully yet. I will just leave a few things that I have found so far (which you probably already know):
- The css of the filter box seems to no longer be working properly. The input boxes for the ranges are no longer aligned
- The mapping of bit number and bit name has not yet been adapted to work with the reversed direction of the `SimStatusBitFieldDisplayValues` mapping
- I am not sure if you currently keep the `filters/mcparticles.js` around to keep things working or if it can be removed already now, since it will live in `filters/collections/mcparticles.js` after this PR

From my testing it looks like the re-arranging of the filtered objects seems to be working now, but it's hard to say from my testfile which is a bit too large to visually grok this.

**Comment by brauliorivas** [comment]
> * am not sure if you currently keep the `filters/mcparticles.js` around to keep things working or if it can be removed already now, since it will live in `filters/collections/mcparticles.js` after this PR

Yes, I'll keep it to guide myself while re-doing filters. Then, I'll remove it.

**Comment by tmadlener** [comment]
A few more comments from my side that are related to this PR. I have opened a few issues which I think are unrelated to changes in this PR and it should be possible to deal with them after this has been finished.

- The `Filter` "window" / overlay is now visible on the landing page. Is it possible to remove it again from there?
image
- The MCParticle filters seem to be partially broken (?). I can filter e.g. on `momentum`, but if I try to filter on the generator or simulation status nothing is happening (unless I am too impatient, but I think this was once almost instantaneous, which is why I am now wondering).
- I don't think this PR triggered this, but I can't seem to easily reproduce this on the main branch: I managed to get MCParticle overlays getting stuck
image
Given that I do not really now how I got to this state on this PR preview, and I also can't reproduce it anywhere, consider this mainly as "we have seen this before once", but don't really act on this until we have a more solid reproducer.

**Comment by kjvbrt** [comment]
image

Pop up does not dissapear

**Comment by brauliorivas** [comment]
Hi @kjvbrt, you may check filters for `MCParticle`. Compared to the previous version of filters, when reconnecting, now works as specified in #53. Also, feature #72 was added. For the moment you may check this PR, but I still don't plan to merge it because I have to add reconnecting functions for other types (but I think these are easier develop).

**Comment by kjvbrt** [comment]
Hi @brauliorivas, Nice work so far :)

Few things I observed:
* The filtered infoboxes fly all over :)
    I mean, it is hard to find them. Can they be always put into the horizontal middle of the screen?
    In case of small number of boxes they stay on the left side, and if there are many they will go to the far right.
    It looks to me that you need to move the view-port (or screen) where the boxes are (maybe you will need to also enlarge the drawing area to be at least as wide as the screen in the case of small number of boxes)
    Also, there should be scrollbars, so people will understand that they should/can scroll.
    This should be the result after strict filter:
    image
* Can you put the toggle for the "Show PDG" on the right side?
* Can you reduce usage of scrollbar in the filters menu? I mean, to reduce the gap between screen bottom and bottom of the menu box.
* My particles don't fit into the infoboxes:
    image

**Comment by kjvbrt** [comment]
Hi @brauliorivas, looks good :)

The MCParticle filter seems to be working nicely now :)

* Can you reduce the usage of scrollbars also for the views selector?
* The toggle could be a little more to the right (aligned with the arrow down of the view selector) :)
* The size of the font for PDG ID/particle symbol varies depending on the sign/sub(sup)scripts, can there be a fixed size for the number/symbol itself?
    Now the numbers/symbols look wonky:
    image
    image

* When I change the size of the window the scroll bars are not adapted.
